sdk/communication/Azure.Communication.Rooms/tests/RoomsClient/RoomsClientTest.cs

```diff
@@ -150,6 +150,40 @@ public void GetRoomShouldSucceed()
             Assert.AreEqual(expectedRoomResult, actualResponse);
         }
 
+        [Test]
+        public void GetRoomsAsyncShouldSucceed()
+        {
+            Mock<RoomsClient> mockRoomsClient = new Mock<RoomsClient>();
+            AsyncPageable<CommunicationRoom> expectedRoomResult = new Mock<AsyncPageable<CommunicationRoom>>().Object;
```

Comment:
Can we also specify a dummy expectedRoomResult? We can also add some assertions on the actualResponse (first meeting properties). This would allow us test against some of the domain logics in between.

Reply:
Would this test anything additional to what the live tests are already testing?

Reply:
Ah ic, it won't be much benefit in this case. However, a few tests we can consider to add to the mock scenarios to cover a different scenarios, especially on a negative cases. 

1. Request - mocked nextLink.
2. Return - an error code (500 internal server error).
3. Return -  mocked Null value and available next link 
4. Return - mocked null value and null next link

Reply:
Yea, it's a good point, I just don't know if it'll be possible to test the next link specifically because as we chatted, the next link is internalized as part of the AsyncPagable structure in SDKs. I'll look into this a bit more and address in another PR if I am able to mock this so we don't block this current PR if that is okay :)